modules/hal_nordic/nrfx/Kconfig

```diff
@@ -9,62 +9,60 @@ menu "nrfx drivers"
 
 config NRFX_ADC
 	bool "ADC driver"
-	depends on HAS_HW_NRF_ADC
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_ADC))
```

Comment:
I think it should be drivers that select HAL options, see for example ST/GD:

https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_gigadevice/Kconfig#L230-L234

Then, the UART driver just does:

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/Kconfig.gd32#L7-L15

This results in:

- HAL modules being compiled as needed by drivers
- Drivers being automatically selected based on DT status

Reply:
ok, got it, thanks for the explanation